File: jquery.realperson.php
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN"
     "http://www.w3.org/TR/html4/strict.dtd">
<?php
function rpHash($value) {
	$hash = 5381;
	$value = strtoupper($value);
	for($i = 0; $i < strlen($value); $i++) {
		$hash = (($hash << 5) + $hash) + ord(substr($value, $i));
	}
	return $hash;
}
?>
<html>
<head>
<title>jQuery Real Person Submission</title>
<style type="text/css">
.accepted { padding: 0.5em; border: 2px solid green; }
.rejected { padding: 0.5em; border: 2px solid red; }
</style>
</head>
<body>
<h1>jQuery Real Person Submission</h1>
<?php
if (rpHash($_POST['defaultReal']) == $_POST['defaultRealHash']) {
?>
<p class="accepted">You have entered the "real person" value correctly and the form has been processed.</p>
<?php
} else {
?>
<p class="rejected">You have NOT entered the "real person" value correctly and the form has been rejected.</p>
<?php
}
?>
</body>
</html>

File: jquery.realperson.rb
class FormController < BaseController

	# POST /form
	# POST /form.json
	def create
		realPerson = params[:realPerson]
		realPersonHash = params[:realPersonHash]
		
	    if !realPerson.nil? && !realPersonHash.nil? && Integer(rpHash(realPerson)) == Integer(realPersonHash)
			
			# Accepted
		else 
			# Rejected
		end
		
	end

private 

	def rpHash (defaultReal)
		hash = 5381
		if !defaultReal.nil?
			defaultReal.upcase!
			defaultReal.length.times{ |i| hash = ((shift_32 hash, 5) + hash) + defaultReal[i].ord }
		end
		return hash
	end
	
	def shift_32 x, shift_amount
	  shift_amount &= 0x1F
	  x <<= shift_amount
	  x &= 0xFFFFFFFF 
	
	  if (x & (1<<31)).zero?
	   x
	  else
	   x - 2**32
	  end
	end

end